packages/core/src/expression-builders/literal.ts

```diff
@@ -5,7 +5,7 @@ import { BaseSqlExpression } from './base-sql-expression.js';
  */
 export class Literal extends BaseSqlExpression {
   /** this (type-only) brand prevents TypeScript from thinking Cast is assignable to Literal because they share the same shape */
```

Comment:
```suggestion
  /** this (type-only) brand prevents TypeScript from thinking Cast is assignable to Literal because they share the same shape */
```
Just to check; the idea was there already to differentiate this, but making it private was not enough?

Reply:
The type only property didn't exist on the base class `BaseSqlExpression` which means that Typescript is unable to distinguish it between any other class.

For example, it its current state the below is possible without any type errors.
```ts
  await sequelize.query('test'); // this is valid as query accepts string | BaseSqlExpression
  await sequelize.query(/test/); // this is invalid, but no type error is shown
  await sequelize.query(32); // this is invalid, but no type error is shown
```

By setting the type value on the base class and having it protected so child classes can override it with their own unique value, ensures that the classes are unique and will show errors when invalid input values are used.

It might actually be better to have a private symbol for the classes similar to what the DataTypes have as this ensure the class is unique.

Reply:
A private symbol indeed sounds better, is that something you can change in this PR?